File: src/test/java/com/alibaba/ttl/testmodel/Call.java
package com.alibaba.ttl.testmodel;

import com.alibaba.ttl.TransmittableThreadLocal;
import com.alibaba.ttl.Utils;

import java.util.Map;
import java.util.concurrent.Callable;
import java.util.concurrent.ConcurrentMap;

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
public class Call implements Callable<String> {
    public final String tag;
    private ConcurrentMap<String, TransmittableThreadLocal<String>> ttlInstances;

    public Call(String tag, ConcurrentMap<String, TransmittableThreadLocal<String>> ttlInstances) {
        this.tag = tag;
        this.ttlInstances = ttlInstances;
    }

    public volatile Map<String, Object> captured;

    @Override
    public String call() {
        captured = Utils.modifyTtlInstances(tag, ttlInstances);
        return "ok";
    }
}

File: ttl2-compatible/src/test/java/com/alibaba/ttl/testmodel/Call.kt
package com.alibaba.ttl.testmodel

import com.alibaba.createChildTtlInstancesAndModifyParentTtlInstances
import com.alibaba.ttl.TransmittableThreadLocal
import java.util.concurrent.Callable
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.ConcurrentMap

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
class Call(private val tag: String, private val ttlInstances: ConcurrentMap<String, TransmittableThreadLocal<String>> = ConcurrentHashMap()) : Callable<String> {

    lateinit var copied: Map<String, String>

    val isCopied: Boolean
        get() = ::copied.isInitialized

    override fun call(): String {
        copied = createChildTtlInstancesAndModifyParentTtlInstances(tag, ttlInstances)
        return "ok"
    }
}
